app/views/sp_components/show.html.erb

```diff
@@ -73,6 +73,23 @@
   <%= render "services/list/services_table" %>
 <% end %>
 
-<%= link_to t('components.add_service'),
-    polymorphic_url([:new, @component, :service], component: @component),
-    class: 'govuk-button', data: { module: "govuk-button" } %>
+<% if !@available_services.empty? %>
```

Comment:
~if !@available_services.empty?~ => `unless @available_services.empty?`

Reply:
Grr!  I went to StackOverflow to find out what the most canonical way to write this was, and they were all just saying "don't use array.any() to do this", and the only one who gave an actual answer said "use !array.empty?".  Your suggestion definitely looks better.